DQM/L1TMonitor/interface/L1TObjectsTiming.h

```diff
@@ -0,0 +1,114 @@
+#ifndef DQM_L1TMonitor_L1TObjectsTiming_h
+#define DQM_L1TMonitor_L1TObjectsTiming_h
+
+// System include files
+#include <memory>
+#include <vector>
+
+#include "DataFormats/L1Trigger/interface/Muon.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+#include "DataFormats/L1Trigger/interface/Tau.h"
+
+#include "DataFormats/L1TGlobal/interface/GlobalAlgBlk.h"
+#include "DataFormats/L1TGlobal/interface/GlobalExtBlk.h"
+#include "DataFormats/L1Trigger/interface/BXVector.h"
+#include "L1Trigger/L1TGlobal/interface/L1TGlobalUtil.h"
+
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+
+class L1TObjectsTiming : public DQMEDAnalyzer {
+
+ public:
+
+  L1TObjectsTiming(const edm::ParameterSet& ps);
+  virtual ~L1TObjectsTiming();
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+ protected:
+
+  virtual void dqmBeginRun(const edm::Run&, const edm::EventSetup&) override;
+  virtual void beginLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup&) override;
+  virtual void bookHistograms(DQMStore::IBooker&, const edm::Run&, const edm::EventSetup&) override;
+  virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+ private:  
+
+  edm::EDGetTokenT<l1t::MuonBxCollection> ugmtMuonToken;
+  edm::EDGetTokenT<l1t::JetBxCollection> stage2CaloLayer2JetToken_;
+  edm::EDGetTokenT<l1t::EGammaBxCollection> stage2CaloLayer2EGammaToken_;
+  edm::EDGetTokenT<l1t::TauBxCollection> stage2CaloLayer2TauToken_;
+  edm::EDGetTokenT<l1t::EtSumBxCollection> stage2CaloLayer2EtSumToken_;
+
+  edm::EDGetTokenT<GlobalAlgBlkBxCollection> l1tStage2uGtProducer_; // input tag for L1 uGT DAQ readout record
+
+  std::string monitorDir;
+  bool verbose;
+
+  // To get the algo bits corresponding to algo names
+  l1t::L1TGlobalUtil* gtUtil_;
+
+  // For the timing histograms
+  int algoBitFirstBxInTrain_;
+  int algoBitLastBxInTrain_;
+  const std::string algoNameFirstBxInTrain_;
+  const std::string algoNameLastBxInTrain_;
+
+
+  const unsigned int bxrange = 5; //this is the out bx range
```

Comment:
Initialise the variable in the initializer list of the class constructor instead of here.